src/fides/api/service/connectors/base_erasure_email_connector.py

```diff
@@ -0,0 +1,137 @@
+from typing import Any, Dict, List
+
+from sqlalchemy.orm import Session
+
+from fides.api.models.connectionconfig import ConnectionConfig, ConnectionType
+from fides.api.models.policy import Rule
+from fides.api.models.privacy_request import (
+    ExecutionLog,
+    ExecutionLogStatus,
+    PrivacyRequest,
+)
+from fides.api.schemas.connection_configuration.connection_secrets_email import (
+    AdvancedSettings,
+    BaseEmailSchema,
+    IdentityTypes,
+)
+from fides.api.schemas.messaging.messaging import (
+    ErasureRequestBodyParams,
+    MessagingActionType,
+)
+from fides.api.schemas.policy import ActionType
+from fides.api.schemas.redis_cache import Identity
+from fides.api.service.connectors.base_email_connector import (
+    BaseEmailConnector,
+    get_email_messaging_config_service_type,
+    get_org_name,
+)
+from fides.api.service.messaging.message_dispatch_service import dispatch_message
+from fides.config import get_config
+
+CONFIG = get_config()
+
+ERASURE_EMAIL_CONNECTOR_TYPES = [
+    ConnectionType.generic_erasure_email,
+    ConnectionType.attentive,
+    ConnectionType.dynamic_erasure_email,
+]
+
+
+class BaseErasureEmailConnector(BaseEmailConnector):
```

Comment:
Most of the logic here was directly taken from the GenericEmailErasureConnector, just made the typing a bit more flexible so we can inherit from it in both cases